tinygrad/ops.py

```diff
@@ -164,6 +165,13 @@ def __init__(self, jit_cache: List[JitItem], input_rawbuffers: Dict[Union[int, s
         if a in [v for v in input_rawbuffers.values()]:
           self.input_replace[(j,i)] = [k for k,v in input_rawbuffers.items() if v == a][0]
     assert set(self.input_replace.values()) == set(input_rawbuffers.keys()), "some input tensors not found"
+
+    # Record info about updatable jit entries.
+    def dim_has_vars(dim): return dim and any(isinstance(x, Node) for x in dim)
+    self.updatable_launch_dims: List[int] = [j for j,ji in enumerate(jit_cache) if isinstance(ji.prg, CompiledASTRunner) and (dim_has_vars(ji.prg.global_size) or dim_has_vars(ji.prg.local_size))]
+    self.updatable_var_vals: List[int] = [j for j,ji in enumerate(jit_cache) if ji.prg.vars]
+    self.updatable_inputs: List[int] = [x[0] for x in self.input_replace.keys()]
```

Comment:
I'm confused about this last one. The other two should have names that reflect that they are kernel indexes